src/test/java/org/broadinstitute/hellbender/engine/dataflow/VariantTest.java

```diff
@@ -0,0 +1,62 @@
+package org.broadinstitute.hellbender.engine.dataflow;
+
+import htsjdk.variant.variantcontext.VariantContext;
+import htsjdk.variant.vcf.VCFCodec;
+import org.broadinstitute.hellbender.engine.FeatureDataSource;
+import org.broadinstitute.hellbender.engine.SkeletonVariant;
+import org.broadinstitute.hellbender.engine.Variant;
+import org.broadinstitute.hellbender.engine.VariantAdapter;
+import org.broadinstitute.hellbender.utils.SimpleInterval;
+import org.broadinstitute.hellbender.utils.test.BaseTest;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+import java.io.File;
+import java.util.*;
+
+
+public class VariantTest extends BaseTest {
+    private static final String FEATURE_DATA_SOURCE_TEST_DIRECTORY = publicTestDir + "org/broadinstitute/hellbender/engine/";
+    private static final File QUERY_TEST_VCF = new File(FEATURE_DATA_SOURCE_TEST_DIRECTORY + "feature_data_source_test.vcf");
+
+    @Test
+    public void testVariantAdapter() {
+        // The test suite for reading in VCF files is FeatureDataSourceUnitTest.
+        FeatureDataSource<VariantContext> featureSource = new FeatureDataSource<>(QUERY_TEST_VCF, new VCFCodec());
+        List<Variant> variantList = new ArrayList<>();
+        for (VariantContext feature : featureSource) {
+            VariantAdapter va = VariantAdapter.of(feature);
+            variantList.add(va);
+        }
+        checkVariants(variantList);
+        featureSource.close();
+    }
+
+    private void checkVariants(final List<Variant> variantList) {
+        Set<Variant> variantSet = new HashSet<>();
+        variantSet.add(new SkeletonVariant(new SimpleInterval("", 99, 99), true));
+        variantSet.add(new SkeletonVariant(new SimpleInterval("", 198, 199), false));
+        variantSet.add(new SkeletonVariant(new SimpleInterval("", 199, 199), true));
+        variantSet.add(new SkeletonVariant(new SimpleInterval("", 202, 205), false));
+        variantSet.add(new SkeletonVariant(new SimpleInterval("", 279, 279), true));
+        variantSet.add(new SkeletonVariant(new SimpleInterval("", 283, 285), false));
+        variantSet.add(new SkeletonVariant(new SimpleInterval("", 284, 284), true));
+        variantSet.add(new SkeletonVariant(new SimpleInterval("", 285, 285), true));
+        variantSet.add(new SkeletonVariant(new SimpleInterval("", 998, 998), true));
+        variantSet.add(new SkeletonVariant(new SimpleInterval("", 999, 999), true));
+        variantSet.add(new SkeletonVariant(new SimpleInterval("", 999, 1002), false));
+        variantSet.add(new SkeletonVariant(new SimpleInterval("", 1075, 1075), true));
+
+        // Check that the first handful of variants match what's expected from the file.
+        for (Variant v : variantSet) {
+            Boolean found = false;
+            for (Variant vv : variantList) {
+                if (v.isSnp() == vv.isSnp() && v.getStart() == vv.getStart() && v.getEnd() == vv.getEnd()) {
```

Comment:
Seems like you need to implement `equals()` (and therefore `hashCode()`) in your variant implementations.